### DISCUSSION TITLE
Does top-langs api also count languages from private repos?

### DISCUSSION THREAD

**Discussion by isayakmondal** [anuraghazra/github-readme-stats#1554] [Q&A] [answered]
I've added a private repo with a new language probably 10 hours ago but it still shows the same number of languages. I've changed the `&langs_count=` to 8 but it still shows 7 languages in total.
So I was wondering does this API only counts from public repos?

**Comment by rickstaa** [comment, marked as answer]
@isayakmondal To my knowledge, private contributions for the language card only work on self-hosted vercel instances. For the public vercel instance (i.e. https://github-readme-stats.vercel.app/), it can not work since we do not have access to your private repositories. As a result, we can not count the language bytes. If you want to show languages used in your private repositories I would therefore advise you to set up your vercel instance. A guide on how this is done can be found [here](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance).

**Reply by isayakmondal** [reply]
Yeah I could do that, but I've one question in mind, so if I fork the repo now that fork would be the current instance of the original repo right? So in future if there are bug fixes or any enhancement on the original repo how will I get that? Do I have to PR every time?

**Reply by rickstaa** [reply]
@isayakmondal You are right, a fork uses the state of the upstream repository when it is cloned. It, however, is straightforward to pull upstream changes into your fork. Github even has a GUI button for that (see https://stackoverflow.com/questions/39819441/keeping-a-fork-up-to-date#:~:text=From%20GitHub%3A,compare%20the%20changes%20before%20merging). You can also, for example, use a GitHub action to automatically do this when the upstream changes (e.g. https://github.com/wei/pull).

Lastly, you can also buy a premium vercel subscription and let your vercel instance point directly to the upstream. We ask you to fork the repository because the `max_duration` parameter of the upstream is set to `30` where free accounts need it to be set to `10`.

**Reply by isayakmondal** [reply]
@rickstaa Alright, but if I change the `maxDuration`  in that `vercel.json` file how will the future pull request gonna work? Cause the base of the two repos are not same right? So every time I wanna merge a PR do I have to change `maxDuration` to `10` every time?

**Reply by rickstaa** [reply]
No I have not encountered that problem yet. Github should take care of that and keep the commit in which you changed the max duration. 👍
